src/java/org/apache/cassandra/utils/ByteBufferUtil.java

```diff
@@ -867,4 +867,70 @@ private static boolean startsWith(ByteBuffer src, ByteBuffer prefix, int offset)
 
         return true;
     }
+
+    public static short getShort(byte[] memory, int index)
```

Comment:
@dcapwell I think @bdeggleston has implemented a lot of this in [his work|https://github.com/apache/cassandra/blob/d426d967b4d9a57ab8cee318d667d0f86a94c449/src/java/org/apache/cassandra/utils/ByteArrayUtil.java] on [CASSANDRA-15393](https://issues.apache.org/jira/browse/CASSANDRA-15393). I just want to make sure we don't have two slightly different versions of these utilities hanging around...

Reply:
Looking again, are only `setInt()` and `setLong()` actually used?

Reply:
yeah, this was copy/pasted from netty, I used short before but then saw that quick theories generates garbage for each `.next(rnd)` call so changed the logic to use int/long when I could to avoid allocations.

Reply:
@bdeggleston how does this sound to you, if your patch goes in first i'll switch to your functions, if this goes in first you can switch this logic to yours?  Seems they are basically the same on first inspection.

Reply:
for now ill create the `ByteArrayUtil` class and match signatures.

Reply:
Ok, I pulled in part of the ByteArrayUtil class, and made a few changes

1) all get methods now also have an overload where offset = 0 (before only 1 method had this, now all do)
2) document byte ordering
3) switched from + to |
4) added tests for methods